### ISSUE TITLE
[FEA]Support ArrayFilter

### ISSUE THREAD

**Issue by viadea** [NVIDIA/spark-rapids#10760]
I wish we can support ArrayFilter.
eg:
```
scala>  spark.sql("SELECT filter(array(1, 2, 3), x -> x % 2 == 1)").show
```

Not-supported-messages:
```
      ! <ArrayFilter> filter([1,2,3], lambdafunction(((lambda x#0 % 2) = 1), lambda x#0, false)) cannot run on GPU because GPU does not currently support the operator class org.apache.spark.sql.catalyst.expressions.ArrayFilter
```

**Comment by matthewthekramer** [comment]
Hey folks - apologies if this is the wrong place to ask - but I see that the initialization action for GCP dataproc is still on 24.04 and this change was merged to 24.06. Is it safe to just update the version in the spark-rapids.sh script manually or is there somewhere I can look for more up-to-date spark-rapids.sh scripts? Thanks!
https://github.com/GoogleCloudDataproc/initialization-actions/blob/master/spark-rapids/spark-rapids.sh

**Comment by viadea** [comment]
> Hey folks - apologies if this is the wrong place to ask - but I see that the initialization action for GCP dataproc is still on 24.04 and this change was merged to 24.06. Is it safe to just update the version in the spark-rapids.sh script manually or is there somewhere I can look for more up-to-date spark-rapids.sh scripts? Thanks! https://github.com/GoogleCloudDataproc/initialization-actions/blob/master/spark-rapids/spark-rapids.sh

Normally it will take couple of days to update the init script for dataproc in GCP repo due to approval/CICD needed from GCP side. 
It is updated now.
